vcsclient/bitbucketcloud.go

```diff
@@ -16,21 +15,29 @@ import (
 
 type BitbucketCloudClient struct {
 	vcsInfo VcsInfo
+	url     *url.URL
 }
 
 func NewBitbucketCloudClient(vcsInfo VcsInfo) (*BitbucketCloudClient, error) {
-	err := os.Setenv("BITBUCKET_API_BASE_URL", vcsInfo.ApiEndpoint)
-	if err != nil {
-		return nil, err
-	}
 	bitbucketClient := &BitbucketCloudClient{
 		vcsInfo: vcsInfo,
 	}
+	if vcsInfo.ApiEndpoint != "" {
+		url, err := url.Parse(vcsInfo.ApiEndpoint)
+		if err != nil {
+			return nil, err
+		}
+		bitbucketClient.url = url
```

Comment:
If this value can be computed from `bitbucketClient.ApiEndpoint`, then I suggest we always compute it instead of "caching" the computation result.

Meaning, we should compute the URL in the `buildBitbucketCloudClient` function.

Reply:
Just to avoid having in a struct a field whose value can be computed. A bit like having two fields 'age' and 'date of birth' for a 'person' type.
Anyway, your call.